### PR TITLE
生成跑道

### PR DESCRIPTION
**Pull request by AaronZSAM101**

### CODE REVIEW

File: Generate_RWY.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+latdd=int(str[1:3])
+latmm=int(str[3:5])
+latss=round(float(str[5:]),3)
+longdd=int(str[1:3])
+longmm=int(str[3:5])
+longss=round(float(str[5:]),3)
+EuroScopeFormat=f"{str[0]}N{latdd}.{latmm}.{latss} E{longdd}.{longmm}.{longss}"
+
+with open('rwy.txt', 'w', encoding='gbk') as output_file:
+
```
Comment (sparrowhe) [inline Generate_RWY.py:102]: code format

File: Generate_RWY.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    dic_rwy[tmp]=rwy+data
+for rwy_direction in rwy_direction_data:
+    if rwy_direction[0] in dic_rwy:
+        data=dic_rwy[rwy_direction[0]]
+    else:
+        data=()
+    tmp=rwy_direction[0]
+    rwy_direction=rwy_direction[0:]
+    if tmp in dic_total:
+        dic_total[tmp]=dic_total[tmp][:2]+rwy_direction[1:2]+dic_total[tmp][2:3]+rwy_direction[2:3]+dic_total[tmp][3:5]+rwy_direction[3:5]+dic_total[tmp][5:]
```
Comment (sparrowhe) [inline Generate_RWY.py:50]: use format string

File: Generate_RWY.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+# 过滤不包含 'N'、'E' 的行且包含中文字符的行
+filtered_lines = [line for line in lines if ('N' in line or 'E' in line) and contains_chinese(line)]
+
+# 打开输出文件并将过滤后的行写入
+with open('output.txt', 'w', encoding='utf-8') as output_file:
+    output_file.writelines(filtered_lines)
+
+print("多余数据删除完成")
+
+
```
Comment (sparrowhe) [inline Generate_RWY.py:90]: code format: unnecessary empty lines

File: TEST.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            elif part.startswith("E"):
+                converted_part = convert_coordinates(part[1:], "lon")
+                converted_parts.append(part[0] + converted_part)
+            else:
+                converted_parts.append(part)
+
+        output_line = ' '.join(converted_parts)
+        output_file.write(output_line + "\n")
+
+print("Conversion complete. Output saved in", output_filename)
```
Comment (sparrowhe) [inline TEST.py:34]: leave a new line at the file end

File: convert_test.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+def convert_coordinates(coord_str):
+    degree = int(coord_str[:3])
+    minute = int(coord_str[3:5])
+    seconds = float(coord_str[5:])
+    converted_seconds = round((seconds * 60), 3)
+    return f"{degree:03d}.{minute:02d}.{converted_seconds:06.3f}"
+
```
Comment (sparrowhe) [inline convert_test.py:7]: think about delete this file if it's unnessary

File: output.txt [hunk 6]
```diff
@@ -0,0 +1,290 @@
+18L 36R 179 359 N400522.00 E1163541.31 N400319.55 E1163600.51 ZBAA 北京/首都
```
Comment (sparrowhe) [inline output.txt:1]: add it to gitignore

File: rwy.txt
Comment (sparrowhe) [inline rwy.txt:1]: add it to gitignore

File: Generate_RWY.py (hunk 2 above)
Comment (AaronZSAM101) [reply Generate_RWY.py:50]: 1

File: Generate_RWY.py (hunk 1 above)
Comment (AaronZSAM101) [reply Generate_RWY.py:102]: no

File: TEST.py (hunk 4 above)
Comment (AaronZSAM101) [reply TEST.py:34]: 1

File: Generate_RWY.py (hunk 3 above)
Comment (AaronZSAM101) [reply Generate_RWY.py:90]: 1

File: convert_test.py (hunk 5 above)
Comment (AaronZSAM101) [reply convert_test.py:7]: 1

File: output.txt (hunk 6 above)
Comment (AaronZSAM101) [reply output.txt:1]: 1

File: rwy.txt
Comment (AaronZSAM101) [reply rwy.txt:1]: 1

**Pull request merged**